docs/basics/101-3-modify.rst

```diff
@@ -0,0 +1,48 @@
+.. _modify:
+
+Starting from scratch: Modifying content
+----------------------------------------
+
+So far, we've only added new content to the dataset. And we haven't done
+much to that content up this point, to be honest. Lets see what happens if
+we add content, and then modify it.
+
+For this, in the root of ``DataLad-101``, create a simple text file
+called ``notes.txt``. This will contain all of the notes that you take
+throughout the course.
+
+Pick an editor of your choice and write a short summary of how to create
+a DataLad dataset from scratch. Below is one way to do it, using the
+``echo`` command that comes with any Unix system, and a simple redirection
+with ``>``.
+
+.. runrecord:: _examples/DL-101-m-1
+   :language: console
+   :workdir: dl-101
+   :realcommand: cd DataLad-101 && echo "One can create a new dataset with 'datalad create [--description] PATH'. The dataset is created empty" > notes.txt
+
+   $ echo "One can create a new dataset with 'datalad create [--description] PATH'. The dataset is created empty" > notes.txt
+
+Run ``datalad status`` to confirm that there is new, untracked content:
+
+.. runrecord:: _examples/DL-101-m-2
+   :language: console
+   :workdir: dl-101
+   :realcommand: cd DataLad-101 && datalad status
+
+   $ datalad status
+
+Save this file in its current state:
+
+.. runrecord:: _examples/DL-101-m-3
+   :language: console
+   :workdir: dl-101
+   :realcommand: cd DataLad-101 && datalad save -m "Add notes on datalad create" notes.txt
```

Comment:
@mih, @loj: I'm writing further on this, and I'm undecided on which approach to take:

Currently, I'm creating a small text file and ``datalad save`` this. The repository that gets created is a git-annex v5 repository. Therefore, this files content gets locked, and in order to modify it, I need to introduce datalad unlock and the reason behind locking.

An alternative would be to save the textfile to git with the ``to-git`` flag. This would postpone locked data content, but will make an explanation of the differences between saving and not saving to git necessary.

I believe both ways are possible. Do you have a favorite between the two? (or an alternative?)

Reply:
There are more then two alternatives. Few more:

- a V6 repository that gets put into an adjusted unlocked branch (`git annex adjust --unlock`). This is not the same as running `unlock`, it is more radical, but might also turn out to be the necessary thing to do, in order to avoid foreseeable issues down the line. But it is also something that we have least practical experience with -- which doesn't mean that we shouldn't acquire it.

- apply a (sensible) default configuration to the dataset. This will become relevant later anyways (when going over the YODA principles). For now it would be as simple as

  ```
  datalad create -c text2git targetdirectory
  ```
  at the very beginning. This leads to:

  ```
  % cat targetdirectory/.gitattributes

  * annex.backend=MD5E
  **/.git* annex.largefiles=nothing
  * annex.largefiles=(not(mimetype=text/*))%                                                                                                                                                   
  ```

  and ensures that any text file is tracked directly with Git (not annexed).

Reply:
Thanks! A sensible default configuration currently sounds like a good choice at this moment to me, thx for bringing this option up!